auth-tokens/src/main/java/com/palantir/tokens/auth/AuthHeader.java

```diff
@@ -36,8 +36,8 @@ public abstract class AuthHeader {
      * Takes the string form: "Bearer [token]" and creates a new {@link AuthHeader}.
      */
     public static AuthHeader valueOf(String authHeader) {
-        BearerToken bearerToken =
-                BearerToken.valueOf(authHeader.startsWith("Bearer ") ? authHeader.substring(7) : authHeader);
+        BearerToken bearerToken = BearerToken.valueOf(
+                authHeader.regionMatches(true, 0, "Bearer ", 0, 7) ? authHeader.substring(7) : authHeader);
```

Comment:
Would be nice to describe the case-insensitive match and link rfc 7235 here as well, that way future-us have this context.
```suggestion
                authHeader.regionMatches(/* ignoreCase */ true, 0, "Bearer ", 0, 7) ? authHeader.substring(7) : authHeader);
```

Reply:
I generally prefer to rely on the Git history for these types of things, but happy to add comments in the code if you'd prefer.